membership-common/src/main/scala/com/gu/memsub/subsv2/services/SubscriptionService.scala

```diff
@@ -95,53 +90,47 @@ class SubscriptionService[M[_]: Monad](pids: ProductIds, futureCatalog: => M[Cat
     * @see
     *   https://community.zuora.com/t5/Admin-Settings-Ideas/Get-current-active-subscription-rate-plans/idi-p/19049
     */
-  def get(
-      name: memsub.Subscription.Name,
-      isActiveToday: Boolean = false,
-  )(implicit logPrefix: LogPrefix): M[Option[Subscription]] = {
+  def get(name: memsub.Subscription.Name, isActiveToday: Boolean = false)(implicit logPrefix: LogPrefix): M[Option[Subscription]] = {
 
     val url =
       if (isActiveToday)
         s"subscriptions/${name.get}?charge-detail=current-segment" // (effectiveStartDate <= today’s date < effectiveEndDate).
       else
         s"subscriptions/${name.get}" // FIXME: equivalent to ?charge-detail=last-segment which returns even removed historical charges. We should not have this as default.
 
-    val futureSubJson = rest.get[JsValue](url)(idReads, logPrefix)
-
-    futureSubJson.flatMap { subJson =>
-      futureCatalog.map { catalog =>
-        // FIXME: Why naming indicates multiple subscriptions? There should be only one sub per provided name.
-        val allSubscriptionsForSubscriberName = subJson.flatMap { jsValue =>
-          SubscriptionTransform.getSubscription(catalog, pids)(jsValue).withTrace("getAllValidSubscriptionsFromJson")
-        }
-        allSubscriptionsForSubscriberName.leftMap(error => logger.warn(s"Error from sub service for $name: $error")).toOption
-
-      }
+    for {
+      subJson <- rest.get[JsValue](url)(idReads, logPrefix)
+      catalog <- futureCatalog
+    } yield {
+      val allSubscriptionsForSubscriberName =
+        for {
+          jsValue <- subJson.withLogging(s"get subscription $name from zuora")
+          subscription <- SubscriptionTransform.getSubscription(catalog, pids)(jsValue).withLogging(s"getAllValidSubscriptionsFromJson for $name")
+        } yield subscription
+      allSubscriptionsForSubscriberName.toOption
```

Comment:
most of the code in this file seemed to use nested flatmaps which didn't seem clear so I've rewritten as for comprehensions